packages/appkit/src/client/appkit.ts

```diff
@@ -432,14 +433,24 @@ export class AppKit extends AppKitBaseClient {
       ) {
         try {
           ChainController.state.activeChain = caipNetwork.chainNamespace
-          await this.connectionControllerClient?.connectExternal?.({
-            id: ConstantsUtil.CONNECTOR_ID.AUTH,
-            provider: this.authProvider,
-            chain: networkNamespace,
-            chainId: caipNetwork.id,
-            type: UtilConstantsUtil.CONNECTOR_TYPE_AUTH as ConnectorType,
-            caipNetwork
-          })
+
+          if (namespaceAddress) {
```

Comment:
If I'm already connected to EVM before, then when I switch from EVM -> Solana (solana not connected yet), and Solana -> EVM again (EVM is already connected), we are calling `adapter?.switchNetwork` here